pyproject.toml

```
@@ -38,7 +38,7 @@ dependencies = [
   # direct dependencies
   "build",
   "click >= 8",
-  "pip >= 22.2",
+  "pip >= 23",
```

```suggestion
  "pip >= 23.1",
```

Usually, when bumping pip version we only support the two most recent minor versions. Therefore we can bump the pip up to the `23.1` version.

Could you please also grep all `PIP_VERSION` mentions and check for any obsolete branches?